src/components/TextInput/TextInput.view.tsx

```diff
@@ -64,4 +61,33 @@ const TextInput = ({
   );
 };
 
+export type ControlledTextInputProps<FORM_VALUES extends FieldValues> = {
+  label: string;
+  required?: boolean;
+  control: Control<FORM_VALUES>;
+  name: Path<FORM_VALUES>;
+  error: FieldError | undefined;
+};
+
+export const ControlledTextInput = <FORM_VALUES extends FieldValues>({
+  label,
+  name,
+  required,
+  control,
+  error,
+}: ControlledTextInputProps<FORM_VALUES>) => {
+  return (
+    <Controller<FORM_VALUES>
```

Comment:
I think the default way to handle inputs in react-hook-form is to register them using `register` function, Controller should be used only for external libraries where you can't access ref value

Reply:
@bartoszskwara  I don't know how to solve this in your way, if you had any idea, please write it